File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def search_range(nums, target)
    res = nums.each_with_index.filter_map{|n,i| i if n == target}
    res.empty? ? [-1, -1] : [res.first, res.last]
end

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {
        List<int> res = new List<int>();
        for (int i = 0; i < nums.Length; i++) {
            if (nums[i] == target) {
                res.Add(i);
            }
        }
        
        if (res.Count == 0) {
            return new int[] {-1, -1};
        } else {
            return new int[] {res[0], res[res.Count - 1]};
        }
    }
}
